chain/index/chain-indexing-overview-for-rpc-providers.MD

```diff
@@ -0,0 +1,218 @@
+# ChainIndexer Documentation for RPC Providers
+  
+## Introduction
+
+This document is for RPC providers and node operators who serve RPC requests walking through the configuration changes, migration flow and operations/maintenance work needed to enable, backfill and maintain the [`ChainIndexer`](#chainindexer-indexing-system).
+
+**Note: If you are a Storage Provider or node operator who does not serve RPC requests (i.e, if `Fevm.EnableEthRPC = false`), you can skip this document as the `ChainIndexer` is already disabled by default**. 
+
+## ChainIndexer Config
+### Enablement
+
+The following must be enabled on an RPC node before starting as they are disabled by default:
+
+```toml
+[ChainIndexer]
+# Enable the ChainIndexer. 
+  EnableIndexer = true 
+
+[Fevm]
+# Enable the ETH RPC APIs.
+  EnableEthRPC = true
+
+[Events]
+# Enable the Actor Events APIs.
+  EnableActorEventsAPI = true
+```
+
+You can learn more about these configuration options and other configuration options available for the `ChainIndexer` [here](https://github.com/filecoin-project/lotus/blob/master/documentation/en/default-lotus-config.toml).
+
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained. By default, GC is disabled to preserve all indexed data.
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/blob/master/chain/index/gc.go#L15) GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**default**) disables GC.
+- Any positive value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+(*Example:* if your node is configured to retain 2 days of Filecoin chain state with the Splitstore, set `GCRetentionEpochs` to `retentionDays * epochsPerDay = 2 * 2880 = 5760`).
```

Comment:
Splitstore is based on size target, not days, so we need to help them here. I think on average it works out to be ~2 days by default at the moment so so the recommendation is OK, but they're not going to say "I've configured it for X days" because they won't. Most likely they just turned splitstore on and crossed their fingers because splitstore config is confusing.

"Example: If your node is configured to use splitstore with the default thresholds, you are likely retaining 2-3 days worth of data. So you could set ..."

Zen or Orjan may be able to say something more certain about periods here. Just checking on my mainnet node and I have 2.3 days worth of block history.

Reply:
I've love this config setting to be able to track splitstore rather than be rough like this. But I think you looked into that?

Reply:
@rvagg Yes that is incredibly complex to get right because the Splitstore is a beast with multiple knobs that I do not fully understand (I doubt anyone does). So best not to make any suggestions about something we don't understand well ourselves.

Reply:
ok, but for that reason, _"if your node is configured to retain 2 days of Filecoin chain state with the Splitstore"_ isn't helpful because nobody configures it like this, so you need to say something here to align with how splitstore works

Reply:
Handling in https://github.com/filecoin-project/lotus/pull/12600